distributor/cmd/internal/distributor/distributor_test.go

```diff
@@ -59,6 +66,98 @@ var (
 	}
 )
 
+var helper dbHelper
+
+type dbHelper struct {
+	address string
+	nextDB  uint32
+}
+
+func (h *dbHelper) connect() (*sql.DB, error) {
+	connectString := fmt.Sprintf("root:secret@(%s)/mysql", h.address)
+	return sql.Open("mysql", connectString)
+}
+
+func (h *dbHelper) create(testName string) (*sql.DB, error) {
+	db, err := h.connect()
+	if err != nil {
+		return db, err
+	}
+	dbName := fmt.Sprintf("%s_%d", testName, h.nextDB)
+	h.nextDB++
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", dbName))
+	if err != nil {
+		return nil, err
+	}
+	connectString := fmt.Sprintf("root:secret@(%s)/%s", h.address, dbName)
+	return sql.Open("mysql", connectString)
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		glog.Fatalf("Could not construct pool: %s", err)
+	}
+
+	// uses pool to try to connect to Docker
+	err = pool.Client.Ping()
```

Comment:
nit: this could collapse into the old `if err := ...; err != ...`

Reply:
Done.